projects/dotcms-field-elements/src/components/dot-textfield/dot-texfield.e2e.ts

```diff
@@ -1,140 +1,372 @@
+                });
 
-        expect(element.classList.contains('dot-pristine')).toBe(true);
-        expect(element.classList.contains('dot-untouched')).toBe(true);
-        expect(element.classList.contains('dot-invalid')).toBe(true);
-        expect(await input.getProperty('value')).toBe('');
-    });
+                it('should be invalid, untouched, pristine and required when empty on load', async () => {
+                    element.setProperty('value', '');
+                    await page.waitForChanges();
+                    expect(element).toHaveClasses(dotTestUtil.class.emptyRequiredPristine);
+                });
 
-    it('should mark as disabled when prop is present', async () => {
-        element.setProperty('disabled', true);
-        await page.waitForChanges();
-        expect(await input.getProperty('disabled')).toBe(true);
-    });
+                it('should be invalid, touched, dirty and required when valued is cleared', async () => {
+                    element.setProperty('value', 'a');
+                    await page.waitForChanges();
+                    await input.press('Backspace');
+                    await page.waitForChanges();
+                    expect(element).toHaveClasses(dotTestUtil.class.emptyRequired);
+                });
+            });
+        });
 
-    it('should mark as required when prop is present', async () => {
-        expect(await input.getProperty('required')).toBe(true);
+        describe('without data', () => {
+            it('should be pristine, untouched & valid', async () => {
+                await page.setContent(`<dot-textfield></dot-textfield>`);
+                element = await page.find('dot-textfield');
+                expect(element).toHaveClasses(dotTestUtil.class.empty);
+            });
+        });
     });
 
-    it('should set the default value of regexCheck when the Regular Expression is not valid', async () => {
-        element.setProperty('regexCheck', '[^(<[.\\n]+>)]*l');
-        await page.waitForChanges();
-        expect(await input.getProperty('regexCheck')).toBeUndefined();
-    });
+    describe('@Props', () => {
+        beforeEach(async () => {
+            page = await newE2EPage({
+                html: `<dot-textfield></dot-textfield>`
+            });
+
+            element = await page.find('dot-textfield');
+            input = await page.find('input');
+        });
+
+        describe('value', () => {
+            it('should set value correctly', async () => {
+                element.setProperty('value', 'hi');
+                await page.waitForChanges();
+                expect(await input.getProperty('value')).toBe('hi');
+            });
+            it('should render and not break when is a unexpected value', async () => {
+                element.setProperty('value', { test: true });
+                await page.waitForChanges();
+                expect(await input.getProperty('value')).toBe('[object Object]');
```

Comment:
If we can do here `getAttribute()` we don't need this one, just one one with a string and one with any `falsey` value I think.

Reply:
`value` on `input` can be accessed only with `getProperty`, so there are ok